### PR TITLE
Add `DevicePlatform.fromOs` method for shared use.

### PR DESCRIPTION
**Pull request by kenzieschmoll**

### CODE REVIEW

File: pkgs/unified_analytics/lib/src/enums.dart [hunk 1]
```diff
@@ -219,4 +219,16 @@ enum DevicePlatform {
 
   final String label;
   const DevicePlatform(this.label);
+
+  /// Returns the [DevicePlatform] that matches the provided [os].
+  ///
+  /// Where possible, the value for [os] should be derived from
+  /// 'Platform.operatingSystem' from dart:io.
+  static DevicePlatform fromOs(String os) {
```
Comment (eliasyishak) [inline pkgs/unified_analytics/lib/src/enums.dart:227]: Would it be better to pass the `io.Platform` class instead? Just incase someone passes the wrong string and it always defaults to windows

Comment (kenzieschmoll) [reply pkgs/unified_analytics/lib/src/enums.dart:227]: Is it okay to have a dart:io in this library?

Comment (eliasyishak) [reply pkgs/unified_analytics/lib/src/enums.dart:227]: Ah good call! In that case, can you add to the dart doc what the expected values for the switch expression should be, `macos`, `linux`, or `windows`?

**Comment by kenzieschmoll** [comment]
Discussed offline. Closing this to avoid extra complexity.

**Pull request closed without merging**